module/enrichers.mjs

```diff
@@ -0,0 +1,303 @@
+ * ```[[/check ability=cha dc=@abilities.int.dc]]```
+ * becomes
+ * ```html
+ * <a class="roll-action" data-type="check" data-ability="cha" data-dc="15">
+ *   <i class="fa-solid fa-dice-d20"></i> DC 15 Charisma check
+ * </a>
+ * ```
+ */
+export async function enrichCheck(config, label, options) {
+  config = Object.entries(config).reduce((config, [k, v]) => {
+    const bool = foundry.utils.getType(v) === "boolean";
+    if ( bool && (k in CONFIG.DND5E.enrichmentLookup.abilities) ) config.ability = k;
+    else if ( bool && (k in CONFIG.DND5E.enrichmentLookup.skills) ) config.skill = k;
+    else if ( bool && (k in CONFIG.DND5E.enrichmentLookup.tools) ) config.tool = k;
+    else if ( bool && Number.isNumeric(k) ) config.dc = Number(k);
+    else config[k] = v;
+    return config;
+  }, {});
+
+  let invalid = false;
+
+  const skillConfig = CONFIG.DND5E.enrichmentLookup.skills[config.skill];
+  if ( config.skill && !skillConfig ) {
+    console.warn(`Skill ${config.skill} not found while enriching ${config.input}.`);
+    invalid = true;
+  } else if ( config.skill && !config.ability ) {
+    config.ability = skillConfig.ability;
+  }
+
+  const toolUUID = CONFIG.DND5E.enrichmentLookup.tools[config.tool];
+  const toolIndex = toolUUID ? Trait.getBaseItem(toolUUID, { indexOnly: true }) : null;
+  if ( config.tool && !toolIndex ) {
+    console.warn(`Tool ${config.tool} not found while enriching ${config.input}.`);
+    invalid = true;
+  }
+
+  let abilityConfig = CONFIG.DND5E.enrichmentLookup.abilities[config.ability];
+  if ( config.ability && !abilityConfig ) {
+    console.warn(`Ability ${ability} not found while enriching ${config.input}.`);
+    invalid = true;
+  } else if ( !abilityConfig ) {
+    console.warn(`No ability provided while enriching check ${config.input}.`);
+    invalid = true;
+  }
+
+  if ( config.dc && !Number.isNumeric(config.dc) ) config.dc = simplifyBonus(config.dc, options.rollData ?? {});
+
+  if ( invalid ) return config.input;
+
+  // Insert the icon and label into the link
+  if ( !label ) {
+    const ability = abilityConfig?.label;
+    const skill = skillConfig?.label;
+    const tool = toolIndex?.name;
+    if ( ability && (skill || tool) ) {
+      label = game.i18n.format("EDITOR.DND5E.Inline.SpecificCheck", { ability, type: skill ?? tool });
+    } else {
+      label = ability;
+    }
+    const longSuffix = config.format === "long" ? "Long" : "Short";
+    if ( config.passive ) {
+      label = game.i18n.format(`EDITOR.DND5E.Inline.DCPassive${longSuffix}`, { dc: config.dc, check: label });
+    } else {
+      if ( config.dc ) label = game.i18n.format("EDITOR.DND5E.Inline.DC", { dc: config.dc, check: label });
+      label = game.i18n.format(`EDITOR.DND5E.Inline.Check${longSuffix}`, { check: label });
+    }
+  }
+
+  if ( config.passive ) {
+    const span = document.createElement("span");
+    span.classList.add("passive-check");
+    span.innerText = label;
+    return span;
```

Comment:
Might be nice to factor this out into its own function too like `createPassiveElement` or something. I like how compartmentalised this module is currently.